### PR TITLE
Added version number.

### LINKED ISSUE #429: Add `ids-enterprise-ng` version to page body
**Is your feature request related to a problem? Please describe.**

To help determine the **ids-enterprise-ng** version an application is compiled with, the current version should be added to the page body, like the **ids-enterprise** version currently is.

**Describe the solution you'd like**

The result should be as follows:

```html
<html data-ids-enterprise-ng-version="5.2.1" data-sohoxi-version="4.17.1" ...>
...
</html>
```

The implementation for this could be done as an **APP_INITIALIZER** that modifies the body of the application on start.

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #432: Accessibility - Modals Have the Wrong Role
**Describe the bug**
The modal in NG has the wrong role.

**To Reproduce**
1. Go to http://localhost:4200/modal 
2. Click on modal to open
3. Inspect the DOM
4. You will see `role="alertdialog"` this is only valid for messages not on modals. 
5. http://latest-enterprise.demo.design.infor.com/components/modal/example-index.html

**Expected behavior**
Should have the correct role in this case. Something is wrong in the wrapper.

- [x] Add an Axe check for correct accessibility roles - this isnt possible.  - The role is valid but the context is wrong.

**See also**
https://jira.lawson.com/browse/LMCLIENT-24206

### PR DESCRIPTION
**Pull request by bthharper**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

To help determine the **ids-enterprise-ng** version an application is compiled with, the current version is now added to the html element (if present).

**Related github/jira issue (required)**:

Closes #429.
Closes #432 

**Steps necessary to review your pull request (required)**:

clone the enterprise-ng library

```sh
npm i
npm run build:lib
npm run start
```

Verify that the html element has the correct version number.

### CODE REVIEW

File: projects/ids-enterprise-ng/src/lib/version/version-initializer.module.spec.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { SohoVersionInitializerModule } from './version-initializer.module';
+
+describe('SunSystemsAuthModule', () => {
```
Comment (tmcconechy) [inline projects/ids-enterprise-ng/src/lib/version/version-initializer.module.spec.ts]:
```suggestion
describe('Soho Authorization Module', () => {
```

File: projects/ids-enterprise-ng/src/lib/version/version-initializer.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { TestBed } from '@angular/core/testing';
+
+import { SohoVersionInitializerService } from './version-initializer.service';
+
+describe('SunSystemsI18nInitialiserService', () => {
```
Comment (tmcconechy) [inline projects/ids-enterprise-ng/src/lib/version/version-initializer.service.spec.ts]:
```suggestion
describe('Soho I18n Initialiser Service', () => {
```

File: src/app/modal-dialog/modal-dialog.demo.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export class ModalDialogDemoComponent {
   public closeResult: string;
 
   public title = 'Example Modal Dialog';
-  public isAlert = true;
+  public isAlert = false;
```
Comment (tmcconechy) [inline src/app/modal-dialog/modal-dialog.demo.ts]: https://github.com/infor-design/enterprise-ng/issues/432#issuecomment-483993245 - Does tihs solve this issue or do we need to update the demo as well. If so maybe just update the demo as well.

File: projects/ids-enterprise-ng/src/lib/version/version-initializer.service.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+
+// @todo unable to load the version from a file not included as a module in the package.
+// const version = require('../../../../package.json').version;
+// const version = process.env.npm_package_version;
+const version = '5.2.1';
```
Comment (tmcconechy) [inline projects/ids-enterprise-ng/src/lib/version/version-initializer.service.ts]: Seems to be a few suggestions on https://github.com/angular/angular-cli/issues/11931 you could try

File: package.json [hunk 5]
```diff
@@ -28,19 +28,19 @@
...
-    "@angular/router": "~7.2.9",
+    "@angular/animations": "~7.2.12",
+    "@angular/common": "~7.2.12",
+    "@angular/compiler": "~7.2.12",
+    "@angular/core": "~7.2.12",
+    "@angular/forms": "~7.2.12",
+    "@angular/http": "~7.2.12",
+    "@angular/platform-browser": "~7.2.12",
+    "@angular/platform-browser-dynamic": "~7.2.12",
+    "@angular/router": "~7.2.12",
```
Comment (tmcconechy) [inline package.json]: Seems like latest is 
```
usmvtmcconechy:enterprise-ng tmcconechy$ npx ng update
    We analyzed your package.json, there are some packages to update:

      Name                               Version                  Command to update
     --------------------------------------------------------------------------------
      @angular/cli                       7.3.6 -> 7.3.8           ng update @angular/cli
@angular/core                      7.2.10 -> 7.2.13         ng update @angular/core
```

File: src/app/modal-dialog/modal-dialog.demo.ts (hunk 3 above)
Comment (bthharper) [reply src/app/modal-dialog/modal-dialog.demo.ts]: Yes - this solves the issue in the demo.

**Comment by tmcconechy** [comment]
@btharrper I’m not sure if this should just go as is or if you want me to request changes. I could let it go and add the updates later on those two packages. And if you think the hardcoding of the version has no solution?

**Comment by bthharper** [comment]
@tmcconechy  - my plan is to add a `version.json` file which can be updated by the release scripts, otherwise things may get out of sync.

**Comment by clepore** [comment]
Also @bthharper you should be able to checkout a branch right from the repo, you shouldn't need to use a fork if you don't want to 👍 (up to you).

**Pull request merged**